src/azul/azulclient.py

```diff
@@ -151,10 +151,10 @@ def index(self, catalog: CatalogName, notifications: Iterable[JSON], delete: boo
         with ThreadPoolExecutor(max_workers=self.num_workers, thread_name_prefix='pool') as tpe:
 
             def attempt(notification, i):
-                log_args = (indexer_url.url, notification, i)
+                log_args = (str(indexer_url), notification, i)
                 try:
                     logger.info("Notifying %s about %s, attempt %i.", *log_args)
-                    self.post_bundle(indexer_url.url, notification)
+                    self.post_bundle(str(indexer_url), notification)
```

Comment:
```suggestion
                    self.post_bundle(str(indexer_url), notification)
```

In this case `post_bundle` could be changed to accept a furl object. It uses `requests` which implicitly applies `str`.

Reply:
Created FIXME #3398.

Reply:
I'd like [this one](https://github.com/DataBiosphere/azul/pull/3370/files#r699719885) to be resolved in this PR.

Reply:
I'd like this to be addressed in this PR, not a later FIXME.

Reply:
I thought that I fixed the change suggested there in the `post_bundle` function: https://github.com/DataBiosphere/azul/commit/d430c8e0033bf900ed06b2427d4d9bef765f1ebe#diff-b73bcd54877f6dc49307665915a153511a7b0ebffc4c6a37ca02d233e5aa54d0L90

Is that not right? or is there a different way to go about it and also do I remove the fixup comment from that function?

Reply:
I wrote 

> I'd like this to be addressed in this PR, not a later FIXME.

but the FIXME comment is still in place. You either resolve an issue and remove the FIXME or you don't and you leave the FIXME in place.

I also wrote

> It uses `requests` which implicitly applies `str`

but you added an explicit invocation of `str`.